exporter/kafkaexporter/authentication.go

```diff
@@ -131,6 +133,15 @@ func configureSASL(config SASLConfig, saramaConfig *sarama.Config) error {
 		return fmt.Errorf(`invalid SASL Mechanism %q: can be either "PLAIN", "AWS_MSK_IAM", "SCRAM-SHA-256" or "SCRAM-SHA-512"`, config.Mechanism)
 	}
 
+	switch config.Version {
+	case 0:
+		saramaConfig.Net.SASL.Version = sarama.SASLHandshakeV0
+	case 1:
+		saramaConfig.Net.SASL.Version = sarama.SASLHandshakeV1
+	default:
+		return fmt.Errorf(`invalid SASL Protocol Version %d: can be either 0 or 1`, config.Version)
```

Comment:
Please add a validation check in Validate()

Reply:
I'm happy to add the check, but I'm a little confused about this. I don't see corresponding validation checks for most of the other keys (e.g. `sasl.mechanism`), so what makes this new one different? I was just following the patterns that I see in the existing codebase. There are only two checks in `Validate()` right now and it's not obvious to me what the criteria is.

Reply:
Fair enough, I added validation checks for the other `Authentication.SASL` keys as well

Reply:
Thank you ;)